### PR TITLE
Secure app from n+1 problems pull request 

### PR DESCRIPTION
**Pull request by theresetuyi**
## Secure app from n+1 problems

- Secure app from n+1 problems.
2sreenshot
3 screenshot
User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/models/integration/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  it 'displays a "Like this post" button' do
+    expect(page).to have_button('Like this post')
+  end
+
+  it 'displays a link to create a new comment' do
+    expect(page).to have_link('Create new comment', href: "/comment/new/#{post.id}")
+  end
+end
```
Comment (Qoosim) [inline spec/models/integration/post_index_spec.rb]: - [ ] Great job for implementing some some integration spec for the views. However, according to the project requirements, some tests are yet to be written. Kindly check below requirements and write their respective tests for them. 

### Project Requirements.
post-index-rqts

File: spec/models/integration/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  it 'displays the comment each commentor left' do
+    comment_text = 'This is a comment.'
+    Comment.create(author: @user, post: @post, text: comment_text)
+    visit post_path(@post)
+    expect(page).to have_content("Comment author: #{user_name}")
+    expect(page).to have_content("Comment content: #{comment_text}")
+  end
+end
```
Comment (Qoosim) [inline spec/models/integration/post_show_spec.rb]: - [ ] Kindly complete the remaining test here too.

### Project requirements.
post-show-rqt

File: spec/models/integration/user_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      follow_redirect!
+      expect(response).to redirect_to(user_path(user_one))
+    end
+  end
+
+  it 'has a title of "List all users"' do
+    get users_path
+    expect(response.body).to include('<h1>List all users</h1>')
+  end
+end
```
Comment (Qoosim) [inline spec/models/integration/user_index_spec.rb]: - [ ] Kindly complete the remaining test for this too.

### Project requirement

user-index

File: spec/models/integration/user_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  scenario "clicking on a user's post redirects to the post's show page" do
+    visit user_path(user)
+
+    posts.each do |post|
+      click_link(post.title)
+      expect(current_path).to eq(post_path(post))
+    end
+  end
+end
```
Comment (Qoosim) [inline spec/models/integration/user_show_spec.rb:42]: - [ ] This one too.

### Project requirement
user-show

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [OPTIONAL] Some of the previously written tests are failing. Kindly attend to them.

prev-test-1

prev-tests-2
  

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @theresetuyi 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**